File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organisation/event/OrganizationEventDto.java
package de.helfenkannjeder.helfomat.api.organisation.event;

import com.fasterxml.jackson.annotation.JsonTypeInfo;
import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;

/**
 * @author Valentin Zickner
 */
@SuppressWarnings({"WeakerAccess", "CanBeFinal"})
@JsonTypeInfo(
    use = JsonTypeInfo.Id.MINIMAL_CLASS,
    property = "type"
)
public abstract class OrganizationEventDto {
    private OrganisationId organisationId;

    protected OrganizationEventDto() {
    }

    public OrganizationEventDto(OrganisationId organisationId) {
        this.organisationId = organisationId;
    }

    public OrganisationId getOrganisationId() {
        return organisationId;
    }

}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import com.fasterxml.jackson.annotation.JsonTypeInfo
import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
@JsonTypeInfo(use = JsonTypeInfo.Id.MINIMAL_CLASS, property = "type")
interface OrganizationEventDto {

    val organizationId: OrganizationId

    fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T

}